data/data/vsphere/main.tf

```diff
@@ -25,10 +32,20 @@ data "vsphere_network" "network" {
 }
 
 data "vsphere_virtual_machine" "template" {
-  name          = var.vsphere_template
+  name          = vsphereprivate_import_ova.import.name
   datacenter_id = data.vsphere_datacenter.datacenter.id
 }
 
+resource "vsphereprivate_import_ova" "import" {
+  name       = "${var.cluster_id}-rhcos"
```

Comment:
do the machine objects need to match up with this name? If they do maybe add that to the commit message.